clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -530,19 +530,12 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
             // MemberEpoch - always sent
             data.setMemberEpoch(membershipManager.memberEpoch());
 
-            // InstanceId - only sent if has changed since the last heartbeat
-            // Always send when leaving the group as a static member
-            membershipManager.groupInstanceId().ifPresent(groupInstanceId -> {
-                if (!groupInstanceId.equals(sentFields.instanceId) || membershipManager.memberEpoch() == ConsumerGroupHeartbeatRequest.LEAVE_GROUP_STATIC_MEMBER_EPOCH) {
-                    data.setInstanceId(groupInstanceId);
-                    sentFields.instanceId = groupInstanceId;
-                }
-            });
+            // InstanceId - always send when leaving the group as a static member
+            membershipManager.groupInstanceId().ifPresent(data::setInstanceId);
```

Comment:
I updated instance ID to be send it all the time when present. I think the last information was that GC relies on this

Reply:
yes, that's the current expectation, so looks good. Let's just update the comment

Reply:
Ah, I didn't see that one, so the change will go away with a merge